api/src/modules/import-data/sourcing-data/import.service.ts

```diff
@@ -109,6 +113,21 @@ export class SourcingRecordsImportService {
       );
 
       await this.sourcingLocationService.save(geoCodedSourcingData);
+
+      const sourcingRecords: SourcingRecord[] = await this.sourcingRecordService.findAllUnpaginated();
+      this.logger.log(
+        `Generating indicator records for ${sourcingRecords.length} sourcing records`,
+      );
+
+      await Promise.all(
+        sourcingRecords.map(async (sourcingRecord: SourcingRecord) => {
+          await this.indicatorRecordsService.calculateImpactValue(
+            sourcingRecord,
+          );
+        }),
+      );
+
+      this.logger.log('Indicator records generated');
```

Comment:
Yes this looks generally right. Calculate the Impact Records async for all new Sourcing Records created.

Reply:
Theoretically this could/should all be wrapped in a transaction?